### PR TITLE
update includes to use non-entry point headers from detail subdir

### PR DESCRIPTION
**Pull request by dirk-thomas**
Related to ros2/rosidl#461.

### CODE REVIEW

File: rclpy/src/rclpy/_rclpy.c [hunk 1]
```diff
@@ -25,7 +25,7 @@
 #include <rcl/remap.h>
 #include <rcl/time.h>
 #include <rcl/validate_topic_name.h>
-#include <rcl_interfaces/msg/parameter_type__struct.h>
+#include <rcl_interfaces/msg/detail/parameter_type__struct.h>
```
Comment (dirk-thomas) [inline rclpy/src/rclpy/_rclpy.c:28]: As an alternative this could also just include `rcl_interfaces/msg/parameter_type.h`?

Comment (jacobperron) [reply rclpy/src/rclpy/_rclpy.c:28]: I agree. I think the "detail" namespace should discourage users (including `rclpy`) from using those headers directly.

Comment (dirk-thomas) [reply rclpy/src/rclpy/_rclpy.c]: Changed in cea1736f4ede0610b16298d85e0b8fa6125efeaa.

**Comment by jacobperron** [comment]
We should update the PR title.

**Pull request merged**